### PR TITLE
Feature/needs UI

### PR DESCRIPTION
**Pull request by jackcasstlesjones**
# What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Styling
- [ ] Build
- [ ] Chore

## Description

- Needs feature UI
- User going through the Needs questions now updates the `mood` value in that Need's db document
- User can navigate forward and back (and quit) through the modal if they make a mistake

## Screenshots

Screenshot 2024-12-11 at 18 34 55
Screenshot 2024-12-11 at 18 34 39
Screenshot 2024-12-11 at 18 34 42
Screenshot 2024-12-11 at 18 34 47
Screenshot 2024-12-11 at 18 34 51

## UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

- [ ] Semantic HTML implemented?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

_Please aim to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Delete branch after merge?

- [ ] Yes
- [ ] No

## What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: src/app/needs/questions/page.tsx
Comment (maxitect) [inline src/app/needs/questions/page.tsx:1]: delete please

File: src/lib/db/DatabaseManager.ts [hunk 1]
```diff
@@ -167,6 +170,44 @@ class DatabaseManager {
...
+        .findOne({ selector: { id: docId } })
+        .exec();
+
+      if (!document) {
+        throw new Error(
+          `Document with ID '${docId}' not found in collection '${collectionName}'`
+        );
+      }
+
+      // Await the patch operation
```
Comment (maxitect) [inline src/lib/db/DatabaseManager.ts:198]:
```suggestion
```

File: src/lib/db/DatabaseManager.ts [hunk 2]
```diff
@@ -167,6 +170,44 @@ class DatabaseManager {
...
+      const updatedDocument = await document.patch({
+        [field]: update,
+      });
+
+      // Log after the update is complete
```
Comment (maxitect) [inline src/lib/db/DatabaseManager.ts:203]:
```suggestion
```

File: public/mockNeeds.json
Comment (maxitect) [inline public/mockNeeds.json:1]: delete please

**Review by maxitect — CHANGES_REQUESTED** [review]
Pending some deletions, this is good to go, thanks Jack - 1259 lines of code mamma mia!

File: public/mockNeeds.json
Comment (jackcasstlesjones) [reply public/mockNeeds.json:1]: done

**Comment by jackcasstlesjones** [comment]
> Pending some deletions, this is good to go, thanks Jack - 1259 lines of code mamma mia!

454 after deleting that json file lol

**Review by maxitect — APPROVED** [review]
Excellento, thanks!

**Pull request merged**